### ISSUE TITLE
Can’t make ‘nil’ obsolete; did you forget a quote mark?

### ISSUE THREAD

**Issue by 99702** [emacs-evil/evil#1839]
Hello , I'm using emacs 30.0.50 version . I cannot run it 

(use-package evil
  :straight t
  :init
  (setq evil-want-keybinding nil))

I get following error , 

```
Debugger entered--Lisp error: (error "Can’t make ‘nil’ obsolete; did you forget a quote mark?")
  error("Can't make `%s' obsolete; did you forget a quote mark?" nil)
  byte-run--constant-obsolete-warning(nil)
  make-obsolete-variable(nil "Semi-exclusivity prevents selecting text + 1st char of next line,\nwithout having to introduce new niche functionality.\nPrefer to set `evil-v$-excludes-newline' to non-nil." "1.15.0")
  byte-code("\301\302\303\304\305DD\306\307\310\311\312\313\314\315\316&\13\210\317\320\321\"\210\301\322\303\304\323DD\324\307\325\311\312\313\326\315\316&\13\210\301\327\303\304\330DD\331\307\332\311\312\313\333\315\316&\13\210\301\334\303\304\335DD\336\307\337\311\312\313\340\315\316&\13\210\301\300\303\304\341DD\342\307\343\311\312&\7\210\344\10\345\346#\210\301\347\303\304\350DD\351\307\343\311\312&\7\210\301\352\303\304\353DD\354\307\343\311\312&\7\210\355\356\357\360\361\362\311\312&\7\210\301\363\303\304\364DD\365\307\343\311\356&\7\210\301\366\303\304\367DD\370\307\371\311\356&\7\210\301\372\303\304\373DD\374\307\375\311\356&\7\210\301\376\303\304\377DD\201@\0\311\312\307\201A\0&\7\207" [evil-want-visual-char-semi-exclusive custom-declare-variable evil-overriding-maps funcall function #f(compiled-function () #<bytecode 0x19800016fe914>) "Keymaps that should override Evil maps.\nEntries have the form (MAP-VAR . STATE), where MAP-VAR is\na keymap variable and STATE is the state whose bindings\nshould be overridden. If STATE is nil, all states are\noverridden." :type (alist :key-type symbol :value-type symbol) :group evil :set #f(compiled-function (var values) #<bytecode -0xdf3a85166554c21>) :initialize evil-custom-initialize-pending-reset add-hook after-load-functions evil-update-pending-maps evil-intercept-maps #f(compiled-function () #<bytecode 0x19800016fe914>) "Keymaps that should intercept Evil maps.\nEntries have the form (MAP-VAR . STATE), where MAP-VAR is\na keymap variable and STATE is the state whose bindings\nshould be intercepted. If STATE is nil, all states are\nintercepted." (alist :key-type symbol :value-type symbol) #f(compiled-function (var values) #<bytecode -0xdf3a873a2053ba1>) evil-motions #f(compiled-function () #<bytecode -0x92f23ac52795e0>) "Non-Evil commands to initialize to motions." (repeat symbol) evil-set-custom-motions evil-visual-newline-commands #f(compiled-function () #<bytecode 0x1980e1c689114>) "Commands excluding the trailing newline of a Visual Line selection.\nThese commands work better without this newline." (repeat symbol) evil-set-visual-newline-commands #f(compiled-function () #<bytecode 0x19800016fe914>) "DEPRECATED.  Will be removed in a future version.\nPrefer to set `evil-v$-excludes-newline' to non-nil.\n\nVisual character selection to beginning/end of line is exclusive.\nIf non nil then an inclusive visual character selection which\nends at the beginning or end of a line is turned into an\nexclusive selection. Thus if the selected (inclusive) range ends\nat the beginning of a line it is changed to not include the first\ncharacter of that line, and if the selected range ends at the end\nof a line it is changed to not include the newline character of\nthat line." boolean make-obsolete-variable "Semi-exclusivity prevents selecting text + 1st char of next line,\nwithout having to introduce new niche functionality.\nPrefer to set `evil-v$-excludes-newline' to non-nil." "1.15.0" evil-v$-excludes-newline #f(compiled-function () #<bytecode 0x19800016fe914>) "If non-nil, `evil-end-of-line' does not move as far as to include\nthe `\n' char at eol. This makes `v$' consistent with `$' used as a\nmotion (e.g. `v$y' is consistent with `y$' in normal state)." evil-text-object-change-visual-type #f(compiled-function () #<bytecode 0x19800016fe5d4>) "Text objects change the current visual state type.\nIf non-nil then a text-object changes the type of the visual state to\nits default selection type (e.g. a word object always changes to\ncharwise visual state). Otherwise the current visual state type is\npreserved." custom-declare-group evil-cjk nil "CJK support" :prefix ...] 12)
  require(evil-vars)
  byte-code("\302\303!\210\302\304!\210\302\305!\210\302\306!\210\302\307!\210\302\310!\210\302\311!\210\302\312!\210\302\313!\210\302\314!\210\302\315!\210\302\316!\210\302\317!\210\10\203<\0\302\320!\210\11\203D\0\302\321!\210\322\323!\210\324\325!\207" [evil-want-integration evil-want-keybinding require evil-vars evil-common evil-core evil-states evil-repeat evil-macros evil-search evil-ex evil-digraphs evil-types evil-commands evil-jumps evil-maps evil-integration evil-keybindings run-hooks evil-after-load-hook provide evil] 2)
  require(evil nil t)
  (not (require 'evil nil t))
  (if (not (require 'evil nil t)) (display-warning 'use-package (format "Cannot load %s" 'evil) :error))
  (progn (condition-case err (setq evil-want-keybinding nil) ((debug error) (funcall use-package--warning29 :init err))) (if (not (require 'evil nil t)) (display-warning 'use-package (format "Cannot load %s" 'evil) :error)))
  (condition-case err (progn (condition-case err (setq evil-want-keybinding nil) ((debug error) (funcall use-package--warning29 :init err))) (if (not (require 'evil nil t)) (display-warning 'use-package (format "Cannot load %s" 'evil) :error))) ((debug error) (funcall use-package--warning29 :catch err)))
  (progn (straight-use-package 'evil) (defvar use-package--warning29 #'(lambda (keyword err) (let ((msg (format "%s/%s: %s" ... keyword ...))) (display-warning 'use-package msg :error)))) (condition-case err (progn (condition-case err (setq evil-want-keybinding nil) ((debug error) (funcall use-package--warning29 :init err))) (if (not (require 'evil nil t)) (display-warning 'use-package (format "Cannot load %s" 'evil) :error))) ((debug error) (funcall use-package--warning29 :catch err))))
  elisp--eval-last-sexp(nil)
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  command-execute(eval-last-sexp)


```

**Comment by axelf4** [comment]
This was fixed by #1833; try updating Evil.

**Comment by 99702** [comment]
Updating works, Thank you
